### ISSUE TITLE
Neovim treesitter crash with nested JSX on ARM

### ISSUE THREAD

**Issue by thatsmydoing** [neovim/neovim#31579]
### Problem

When using treesitter for javascript and specific queries, neovim crashes when looking at deeply nested JSX. This occurs with the stock nvim-treesitter configuration as of https://github.com/nvim-treesitter/nvim-treesitter/commit/a8c082b4f5b3b70bde2ffb08cc112dfd7e515fb3

The problem does not require nvim-treesitter to reproduce though. It's simply the queries defined in nvim-treesitter that causes it.

I've noticed it also only crashes on ARM machines (specifically a Macbook Air M2). On x86_64 machines, the highlighting is broken but it does not crash. Though with my full config it also crashes on x86_64 machines too.

### Steps to reproduce

I've made a [repo](https://github.com/thatsmydoing/tree-sitter-crash) with the setup and steps.

Here's the backtrace from a Fedora 41 VM [bt.txt](https://github.com/user-attachments/files/18137183/bt.txt)

It also crashes on macOS with neovim installed via nix.

### Expected behavior

neovim does not crash

### Nvim version (nvim -v)

v0.10.2

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Fedora 41

### Terminal name/version

alacritty

### $TERM environment variable

alacritty

### Installation

DNF

**Comment by clason** [comment]
Please test the provided releases (on macOS at least). Your backtrace indicates you're using a different tree-sitter version than we do. (Also test the latest nightly, please.)

**Comment by thatsmydoing** [comment]
Nightly does not crash on macOS with the official build nor on Fedora from source, but it does show the weird highlight behavior that I saw on x86_64.

I did also test the macOS official 0.10.2 release and Fedora from source on the v0.10.2 tag and they both crash. Here's the backtrace from Fedora [bt.txt](https://github.com/user-attachments/files/18137320/bt.txt)

> (The crash is in the tree-sitter library, not nvim itself, so this is important.)

I think this is it. I updated nixpkgs again just now and it looks like they've updated tree-sitter to 0.24.3 from 0.22.6 and I don't get the crash anymore there either with neovim 0.10.2 from nix on macOS.

The highlight is still somewhat broken though, only the outermost tags are highlighted while everything inside isn't. Without the offending query, the highlighting works correctly.
